contrib/nydusify/pkg/checker/rule/filesystem.go

```diff
@@ -162,25 +167,26 @@ func (rule *FilesystemRule) walk(rootfs string) (map[string]Node, error) {
 	return nodes, nil
 }
 
-func (rule *FilesystemRule) mountSourceImage() (*tool.Image, error) {
+func (rule *FilesystemRule) mountSourceImage() error {
 	logrus.Infof("Mounting source image to %s", rule.SourceMountPath)
 
 	if err := os.MkdirAll(rule.SourceMountPath, 0755); err != nil {
-		return nil, errors.Wrap(err, "create mountpoint directory of source image")
+		return errors.Wrap(err, "create mountpoint directory of source image")
 	}
 
-	image := &tool.Image{
-		Source: rule.Source,
-		Rootfs: rule.SourceMountPath,
-	}
-	if err := image.Pull(); err != nil {
-		return nil, errors.Wrap(err, "pull source image")
-	}
-	if err := image.Mount(); err != nil {
-		return nil, errors.Wrap(err, "mount source image")
+	layers := rule.SourceParsed.OCIImage.Manifest.Layers
+	for _, l := range layers {
+		reader, err := rule.SourceRemote.Pull(context.Background(), l, true)
```

Comment:
Maybe we need to make concurrency for layer pulling and limit the max concurrency.